src/traces/surface/defaults.js

```diff
@@ -85,7 +87,20 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
         }
     }
 
+    // backward compatibility block
+    if(!surfaceColor) {
+        mapLegacy(traceIn, 'zmin', 'cmin');
+        mapLegacy(traceIn, 'zmax', 'cmax');
+        mapLegacy(traceIn, 'zauto', 'cauto');
+    }
+
     colorscaleDefaults(
-        traceIn, traceOut, layout, coerce, {prefix: '', cLetter: 'z'}
+        traceIn, traceOut, layout, coerce, {prefix: '', cLetter: 'c'}
     );
 };
+
+function mapLegacy(traceIn, oldAttr, newAttr) {
+    if(oldAttr in traceIn && !(newAttr in traceIn)) {
+        traceIn[newAttr] = traceIn[oldAttr];
```

Comment:
@mdtusz @cldougl @chriddyp Are you ok with this way of preserving backward compatibility?

`zmin`, `zmax` and `zauto` are deprecated in favour of `cmin`, `cmax` and `cauto` for surface traces.

The deprecated `z` attributes are mapped to their corresponding `c` attributes in the defaults step so that both `plot` and `restyle` call update the trace objects to the `c` attributes.

If both a `c` and `z` attribute are present, only the `c` attribute is coerced.